public/pages/Correlations/containers/CorrelationsContainer.tsx

```diff
@@ -468,6 +622,803 @@ export class Correlations extends React.Component<CorrelationsProps, Correlation
+                </EuiFlexItem>
+              </EuiFlexGroup>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <ChartContainer chartViewId={'correlated-findings-view'} />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiPanel>
+        <EuiSpacer />
+      </>
+    );
+  };
+
+  private getFilteredTableData = (tableData: CorrelationsTableData[]): CorrelationsTableData[] => {
+    const { logTypeFilterOptions, severityFilterOptions } = this.state;
+    const alertSeverityMap: { [key: string]: string } = {
+      '1': 'critical',
+      '2': 'high',
+      '3': 'medium',
+      '4': 'low',
+      '5': 'informational',
+    };
+
+    const selectedLogTypes = logTypeFilterOptions
+      .filter((item) => item.checked === 'on' && item.visible)
+      .map((item) => item.id);
+
+    const selectedSeverities = severityFilterOptions
+      .filter((item) => item.checked === 'on' && item.visible)
+      .map((item) => item.id.toLowerCase());
+
+    return tableData.filter((row) => {
+      const logTypeMatch = row.logTypes.some((logType) => selectedLogTypes.includes(logType));
+
+      const severityMatch = row.alertSeverity.some((severity) =>
+        selectedSeverities.includes(alertSeverityMap[severity])
+      );
+
+      const searchLower = this.state.searchTerm.toLowerCase();
+      const searchMatch =
+        this.state.searchTerm === '' ||
+        row.correlationRule?.toLowerCase().includes(searchLower) ||
+        row.logTypes.some((type) => type.toLowerCase().includes(searchLower)) ||
+        row.alertSeverity.some((severity) =>
+          alertSeverityMap[severity].toLowerCase().includes(searchLower)
+        ) ||
+        row.findingsSeverity.some((severity) => severity.toLowerCase().includes(searchLower)) ||
+        row.resources.some((resource) => resource.toLowerCase().includes(searchLower));
+      return logTypeMatch && severityMatch && searchMatch;
+    });
+  };
+
+  private debouncedSearch = debounce((searchTerm: string) => {
+    this.setState({ searchTerm }, () => {
+      this.forceUpdate();
+    });
+  }, 300);
+
+  private renderSearchBar = () => {
+    return (
+      <EuiFieldSearch
+        placeholder="Search"
+        value={this.state.searchTerm}
+        onChange={(e) => {
+          e.persist();
+          const searchValue = e.target.value;
+          this.setState({ searchTerm: searchValue });
+          this.debouncedSearch(searchValue);
+        }}
+        fullWidth={true}
+        isClearable={true}
+        compressed={true}
+        aria-label="Search correlations"
+      />
+    );
+  };
+
+  private renderCorrelationsTable = () => {
+    const alertSeverityMap: { [key: string]: string } = {
+      '1': 'critical',
+      '2': 'high',
+      '3': 'medium',
+      '4': 'low',
+      '5': 'informational',
+    };
+
+    const columns: EuiBasicTableColumn<CorrelationsTableData>[] = [
+      {
+        field: 'startTime',
+        name: 'Start time',
```

Comment:
Nitpick: Noticed that the `name` for this column is sentence case whereas the others are title case. I think we typically use sentence case; but the mocks might be using a different format.